pkg/kopia/command/parse_command_output.go

```diff
@@ -430,6 +432,64 @@ func parseKopiaRestoreProgressLine(line string) (stats *RestoreStats) {
 	}
 }
 
+// RestoreResultFromRestoreOutput parses the output of a `kopia restore`
+// line-by-line in search of restore result statistics.
+// It returns nil if no final statistics are found.
+func RestoreResultFromRestoreOutput(
+	restoreStderrOutput string,
+) (stats *RestoreStats) {
+	if restoreStderrOutput == "" {
+		return nil
+	}
+	logs := regexp.MustCompile("[\r\n]").Split(restoreStderrOutput, -1)
+
+	for _, l := range logs {
+		lineStats := parseKopiaRestoreResultLine(l)
+		if lineStats != nil {
+			stats = lineStats
```

Comment:
As far as I understand this condition will override stats until the last matching line even if there are more matching lines before that. Is that intended?